devicemapper/thin_pool_watcher.go

```diff
@@ -85,16 +85,19 @@ func (w *ThinPoolWatcher) Stop() {
 }
 
 // GetUsage gets the cached usage value of the given device.
-func (w *ThinPoolWatcher) GetUsage(deviceId string) (uint64, error) {
+func (w *ThinPoolWatcher) GetUsage(deviceId string) uint64 {
 	w.lock.RLock()
 	defer w.lock.RUnlock()
 
 	v, ok := w.cache[deviceId]
 	if !ok {
-		return 0, fmt.Errorf("no cached value for usage of device %v", deviceId)
+		// TODO: ideally we should keep track of how many times we failed to get the usage for this
+		// device vs how many refreshes of the cache there have been, and return an error e.g. if we've
+		// had at least 1 refresh and we still can't find the device.
+		glog.V(5).Infof("no cached value for usage of device %v", deviceId)
```

Comment:
K, I'll move it back so it doesn't swallow
On Thu, Jul 21, 2016 at 6:52 PM Tim St. Clair <notifications@github.com>
wrote:

> In devicemapper/thin_pool_watcher.go
> <https://github.com/google/cadvisor/pull/1359#discussion_r71801848>:
>
> >  	w.lock.RLock()
> >  	defer w.lock.RUnlock()
> >
> >  	v, ok := w.cache[deviceId]
> >  	if !ok {
> > -		return 0, fmt.Errorf("no cached value for usage of device %v", deviceId)
> > +		// TODO: ideally we should keep track of how many times we failed to get the usage for this
> > +		// device vs how many refreshes of the cache there have been, and return an error e.g. if we've
> > +		// had at least 1 refresh and we still can't find the device.
> > +		glog.V(5).Infof("no cached value for usage of device %v", deviceId)
>
> The problem with consuming the error here is we don't know whether the
> value returned is valid. For instance, is it a problem to set the
> baseUsage to 0 in handler.go when there isn't a cached value?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/cadvisor/pull/1359/files/46c41743878b759623c02a508c7c1439aa2b3307#r71801848>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYgOLZ8wLpvd5AlH1MEvHDq5Km8fuks5qX_gQgaJpZM4JBiYs>
> .
>